### PR TITLE
[STRATCONN-3832] Validate Audience Field inputs while creating Audience

### PR DESCRIPTION
**Pull request by Innovative-GauravKochar**
This Pull request is to add validation of audience field inputs while creating audience.
This PR is also to add personas as separate field in createAudienceInput , removed out of audienceSettings.

Jira ticket:- https://segment.atlassian.net/jira/software/c/projects/STRATCONN/boards/310?assignee=63617339fc0cc7a600b03c6b&selectedIssue=STRATCONN-3832
Related Integrations PR :- https://github.com/segmentio/integrations/pull/2905
## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

Screenshot 2024-06-14 at 5 46 24 PM

Screenshot 2024-06-26 at 3 04 45 PM

Screenshot 2024-06-26 at 3 04 14 PM

Screenshot 2024-06-26 at 3 00 50 PM

### CODE REVIEW

File: packages/destination-actions/src/destinations/yahoo-audiences/index.ts [hunk 1]
```diff
@@ -115,9 +115,8 @@ const destination: AudienceDestinationDefinition<Settings, AudienceSettings> = {
     },
 
     async createAudience(request, createAudienceInput) {
-      const audienceSettings = createAudienceInput.audienceSettings
       // @ts-ignore type is not defined, and we will define it later
-      const personas = audienceSettings.personas as PersonasSettings
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/yahoo-audiences/index.ts:120]: hi @Innovative-GauravKochar - not sure that changing these lines in this way will work. 
It looks like the json for `personas` is located at `createAudienceInput.audienceSettings.personas` and not at `createAudienceInput.personas` . Maybe I'm missing something?

Comment (Innovative-GauravKochar) [reply packages/destination-actions/src/destinations/yahoo-audiences/index.ts:120]: No , We are moving personas from audience Settings and keeping it as seperate field under CreateAudienceInput

File: packages/core/src/destination-kit/index.ts [hunk 2]
```diff
@@ -94,12 +95,15 @@ export type AudienceResult = {
...
 
 export type AudienceMode = { type: 'realtime' } | { type: 'synced'; full_audience_sync: boolean }
+export type Personas = { computation_id: 'string'; computation_key: 'string' }
 
 export type CreateAudienceInput<Settings = unknown, AudienceSettings = unknown> = {
   settings: Settings
 
   audienceSettings?: AudienceSettings
 
+  personas?: Personas
```
Comment (Innovative-GauravKochar) [inline packages/core/src/destination-kit/index.ts:105]: @joe-ayoub-segment  Here we have added personas as seperate field , So now personas would be  accessible like CreateAudienceInput.personas

Comment (joe-ayoub-segment) [reply packages/core/src/destination-kit/index.ts:105]: yes but that's not how the underlying JSON is structured, is it?

Comment (joe-ayoub-segment) [reply packages/core/src/destination-kit/index.ts:105]: Hi Gaurav, would it be possible to see an example of what the underlying JSON looks like?

**Review by joe-ayoub-segment — DISMISSED** [review]

**Review by joe-ayoub-segment — DISMISSED** [review]

**Review by joe-ayoub-segment — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
hi @Innovative-GauravKochar PR deployed.

**Comment by joe-ayoub-segment** [comment]
hi @varadarajan-tw PR deployed

**Comment by varadarajan-tw** [comment]
Hey @joe-ayoub-segment , we reverted this PR right? Just want to double check we didn't deploy it again.

**Comment by joe-ayoub-segment** [comment]
> Hey @joe-ayoub-segment , we reverted this PR right? Just want to double check we didn't deploy it again.

That's correct @varadarajan-tw , AFAIK

**Comment by joe-ayoub-segment** [comment]
It's still being tested in Staging.

**Pull request merged**